src/components/product.js

```diff
@@ -65,6 +75,7 @@ export default class Product extends Component {
    * @param {Object} props - data and utilities passed down from UI instance.
    */
   constructor(config, props) {
+    config.id = normalizeProductId(config).id;
```

Comment:
wow, yeah not mutating the object in the function seems much cleaner. 💯 
as for `storefrontId` instead of `id`, i'm planning on addressing that by converting the id field across the whole library in a separate pr.